canopy_main.ml

```diff
@@ -40,22 +40,22 @@ module Main  (C: CONSOLE) (S: STACKV4) (RES: Resolver_lwt.S) (CON: Conduit_mirag
     in
     let module Store = Canopy_store.Store(C)(Context)(Inflator) in
     let open Canopy_utils in
-    let cache = KeyHashtbl.create 32 in
+    let cache = ref (KeyMap.empty) in
     let open Canopy_config in
     let config = config () in
     let update_atom, atom =
-      Canopy_syndic.atom config Store.last_commit_date cache
+      Canopy_syndic.atom config Store.last_commit_date !cache
```

Comment:
if my intuition for OCaml refs is correct, this will lead to an empty atom feed always (since dereference is done here, whereas it should be delayed till there's stuff in the feed (and thus done in `Canopy_syndic`))